common/ledger/blockledger/fileledger/factory.go

```diff
@@ -97,8 +109,30 @@ func New(directory string, metricsProvider metrics.Provider) (blockledger.Factor
 	if err != nil {
 		return nil, err
 	}
-	return &fileLedgerFactory{
+
+	fileRepo, err := filerepo.New(filepath.Join(directory, "filerepo"), "remove")
+	if err != nil {
+		return nil, err
+	}
+
+	factory := &fileLedgerFactory{
 		blkstorageProvider: p,
 		ledgers:            map[string]*FileLedger{},
-	}, nil
+		removeFileRepo:     fileRepo,
+	}
+
+	files, err := factory.removeFileRepo.List()
+	if len(files) != 0 {
+		for _, fileName := range files {
+			channelID := factory.removeFileRepo.FileToBaseName(fileName)
+			err = factory.Remove(channelID)
+			if err != nil {
+				logger.Errorf("error removing ledger for channel %s: %s", channelID, err.Error())
+				continue
+			}
```

Comment:
If we retry the "Remove" and fail, do we really want to continue? Depending on what the failure was, there is a chance the channel will be detected and initialized, but then since it went through a botched remove, it might fail with panic when the server starts to run it. Or start operating with an inconsistent state.

It seems that in this case the right thing to do it to return an error rather than suppress it. 

I think that the operator can then intervene, analyze the problem and fix it.
For example she can delete the channel's folder and the index manually and restart.

Reply:
I think you are right, as such I've updated this logic to return the error.